src/Psalm/Internal/Type/TemplateStandinTypeReplacer.php

```diff
@@ -1134,7 +1136,21 @@ public static function getMappedGenericTypeParams(
         Atomic $container_type_part,
         ?array &$container_type_params_covariant = null
     ): array {
-        $input_type_params = $input_type_part->type_params;
+        if ($input_type_part instanceof TGenericObject || $input_type_part instanceof TIterable) {
+            $input_type_params = $input_type_part->type_params;
+        } else {
+            $class_storage = $codebase->classlike_storage_provider->get($input_type_part->value);
+
+            $container_class = $container_type_part->value;
+
+            if (strtolower($input_type_part->value) === strtolower($container_type_part->value)) {
+                $input_type_params = $class_storage->getClassTemplateTypes();
+            } elseif (!empty($class_storage->template_extended_params[$container_class])) {
+                $input_type_params = array_values($class_storage->template_extended_params[$container_class]);
+            } else {
+                $input_type_params = array_fill(0, count($class_storage->template_types ?? []), Type::getMixed());
+            }
```

Comment:
This will conflict with https://github.com/vimeo/psalm/pull/8044 when both are merged to master, this one is correct.